packages/ui-library/src/components/token-picker/index.tsx

```diff
@@ -400,11 +400,15 @@ const TokenPicker = ({
     return tokensMatchingSearch;
   }, [tokens, search, allowedPairs, filterByPair, isOnlyAllowedPairs]);
 
-  useEffect(() => {
-    if (search && filteredTokens.length === 0 && validAddressRegex.test(search) && onFetchCustomToken) {
-      onFetchCustomToken(search.toLowerCase() as Address);
-    }
-  }, [search, filteredTokens, onFetchCustomToken]);
+  const onSearchChange = useCallback(
+    (newSearch: string) => {
+      setSearch(newSearch);
+      if (newSearch && validAddressRegex.test(newSearch) && onFetchCustomToken) {
+        onFetchCustomToken(newSearch.toLowerCase() as Address);
```

Comment:
Missing chain id :)

Reply:
I set chainId as optional so its can be calculated directly in the hook by default (using useSelectedNetwork), unless it needs to be specified (as it is in the case for the url param)

Reply:
yeah but this token picker is opened in a specific chainId so we should also forward that so it doesnt find any inconsistencies